lib/modules/auth/services/auth.service.dart

```diff
@@ -35,6 +35,31 @@ class AuthService {
       }
     });
   }
+
+  Future<void> register(
+      String username, String email, DateTime dateOfBirth, String password) {
+    return initializeDio().post('/public/register', data: {
+      'username': username,
+      'email': email,
+      'password': password,
+      'dateOfBirth': dateOfBirth.toIso8601String()
+    }).catchError((dioError) {
+      // no need to parse a result, the method not throwing means the user was registered
+      switch (dioError.runtimeType) {
+        case DioError:
+          switch (dioError.response?.statusCode) {
+            case (404):
+              throw UnauthorizedException();
```

Comment:
What's the logic for Unauthorized being thrown on 404?

Reply:
Just to be consistent with the login method; it will be modified when merging together with [this PR](https://github.com/PC-Chair-Fight/project-frontend/pull/4)